lib/checkCriticalPullRequest.js

```diff
@@ -0,0 +1,137 @@
+const {
+  getAllChangedFiles,
+  hasCriticalLabel,
+  CRITICAL_LABEL,
+} = require('./checkPullRequestJob');
+const { teamLeads } = require('../userWhitelist.json');
+const STORAGE_DIR_PREFIX = 'core/storage/';
+
+/**
+ * @param {import('probot').Octokit.PullsListFilesResponseItem} file
+ */
+const getNewModelsFromFile = (file) => {
+  const newLine = '\n';
+  const changesArray = file.patch.split(newLine);
+  const modelRegex = /(?<addition>\+)(?<classDefinition>class\s)(?<modelName>[a-zA-Z]{2,256})(?<modelSuffix>Model)(?<funDef>\()/;
```

Comment:
We have a similar function to find new Jobs right, we can merge into a single function which accepts a regex and returns new models/files.

Reply:
Yup
Done.